src/app/@ansyn/ansyn/modules/core/components/angle-filter/angle-filter.component.ts

```diff
@@ -0,0 +1,158 @@
+import { Component, ElementRef, HostBinding, HostListener, OnDestroy, OnInit, Renderer2 } from '@angular/core';
+import { getPointByGeometry, toDegrees, toRadians } from '@ansyn/imagery';
+import { ContextMenuShowAngleFilter, IEntryComponent, MapActionTypes } from '@ansyn/map-facade';
+import { Actions, ofType } from '@ngrx/effects';
+import { Store } from '@ngrx/store';
+import { AutoSubscription, AutoSubscriptions } from 'auto-subscriptions';
+import { Point } from 'geojson';
+import { debounceTime, tap } from 'rxjs/operators';
+import { DisplayOverlayAction, SetHoveredOverlayAction, SetMarkUp } from '../../../overlays/actions/overlays.actions';
+import { IOverlay } from '../../../overlays/models/overlay.model';
+import { MarkUpClass } from '../../../overlays/reducers/overlays.reducer';
+
+const WORLD_RADIUS = 6371e3;
+@Component({
+	selector: 'ansyn-angle-filter',
+	templateUrl: './angle-filter.component.html',
+	styleUrls: ['./angle-filter.component.less']
+})
+@AutoSubscriptions()
+export class AngleFilterComponent implements OnInit, OnDestroy, IEntryComponent {
+
+	mapId: string;
+	overlay: IOverlay;
+	_angles: any[];
+	point: Point;
+	@AutoSubscription
+	showAngleFilter$ = this.actions$.pipe(
+		ofType(MapActionTypes.CONTEXT_MENU.ANGLE_FILTER_SHOW),
+		debounceTime(200),
+		tap(this.show.bind(this))
+	);
+
+
+	get angles() {
+		return this._angles;
+	}
+
+	set angles(value: any[]) {
+		const pointLat = this.getLatFromPoint(this.point, true);
+		const pointLong = this.getLongFromPoint(this.point, true);
+		this._angles = value.map((angle) => {
+			const center = getPointByGeometry(angle.footprint);
```

Comment:
don't use the center anywhere
use overlay.sensorLocation -> in the overlayProvider (planet, sentinal, etc.) if unknown put center
check with @carmelorr what to do if not provided